From ee6bba585d53891577089e9dd856eb733d8231f8 Mon Sep 17 00:00:00 2001
From: Rok Mandeljc <rok.mandeljc@gmail.com>
Date: Mon, 3 Feb 2025 21:05:32 +0100
Subject: [PATCH] VHBA: fix building with kernel 6.14-rc1

Kernel 6.14-rc1 simplified the selection of tag allocation policy.
Instead of enum-based value, a boolean is used, and the corresponding
field in the `scsi_host_template` structure was renamed from
`tag_alloc_policy` to `tag_alloc_policy_rr`.

See: https://github.com/torvalds/linux/commit/ce32496
---
 vhba-module/vhba.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff -uraN vhba-module-20240917/vhba.c omv-vhba-module-20240917/vhba.c
--- vhba-module-20240917/vhba.c	2024-09-17 11:48:31.000000000 +0200
+++ omv-vhba-module-20240917/vhba.c	2025-03-05 16:57:10.122974527 +0100
@@ -537,7 +537,9 @@
 #if LINUX_VERSION_CODE < KERNEL_VERSION(3, 19, 0)
     .slave_alloc = vhba_slave_alloc,
 #endif
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 0, 0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 14, 0)
+    .tag_alloc_policy_rr = true,
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(4, 0, 0)
     .tag_alloc_policy = BLK_TAG_ALLOC_RR,
 #endif
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 19, 0) && LINUX_VERSION_CODE < KERNEL_VERSION(4, 4, 0)
